nms/app/views/alarms/components/hooks.ts

```diff
@@ -77,6 +77,57 @@ export function useLoadRules<TRuleUnion>({
   };
 }
 
+/**
+ * Filter a route from routes for a especific ruleName and receiver
+ * @param {*} response from apiUtil.useAlarmsApi
+ * @param {*} initialReceiver string
+ * @param {*} ruleName string
+ * @returns
+ */
+export function filterRouteByRuleName(
+  response: AlertRoutingTree | null | undefined,
+  initialReceiver: string | null | undefined,
+  ruleName: string | null | undefined,
+) {
+  if (!response || typeof response === 'undefined') {
+    throw new Error('apiUtil Response is null or undefined');
+  }
+  if (!ruleName || typeof ruleName === 'undefined') {
+    return response.routes || [];
+  }
+  if (!initialReceiver || typeof initialReceiver === 'undefined') {
+    return response.routes || [];
+  }
+  const {routes} = response;
+  const filteredRoutes = (routes || []).filter((route: AlertRoutingTree) => {
+    const filterToDelete =
+      route.receiver === initialReceiver && route.match?.alertname === ruleName;
+    return route.receiver !== initialReceiver || !filterToDelete;
```

Comment:
`Not (A and B)` is equivalent to `Not A or not B` thus that I believe the logic here could be simplified.

Reply:
Thanks... I applied this modification!

Reply:
The condition on the `route.receiver` is still duplicated.